src/sinks/loki/config.rs

```diff
@@ -36,6 +36,34 @@ impl EncodingConfigMigrator for EncodingMigrator {
     }
 }
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub enum ExtendedCompression {
+    #[serde(rename = "snappy")]
+    Snappy,
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum CompressionConfigAdapter {
+    Original(Compression),
+    Extended(ExtendedCompression),
```

Comment:
```suggestion
    /// Basic compression.
    Original(#[configurable(derived)] Compression),

    /// Loki-specific compression.
    Extended(#[configurable(derived)] ExtendedCompression),
```

Reply:
These need a doc comment and to use `Configurable`-specific helper attributes as part of the on-going https://github.com/vectordotdev/vector/issues/12141 work.

(This will make more sense once you merge in the latest `master` and see how the `config.rs` has since changed.)